docs/_snippets/examples/multi-root-editor.js

```diff
@@ -0,0 +1,378 @@
+				editor.initPlugins()
+					.then( () => editor.ui.init() )
+					.then( () => {
+						const initialData = {};
+
+						// Create initial data object containing data from all roots.
+						for ( const rootName of Object.keys( sourceElements ) ) {
+							initialData[ rootName ] = getDataFromElement( sourceElements[ rootName ] );
+						}
+
+						return editor.data.init( initialData );
+					} )
+					.then( () => editor.fire( 'ready' ) )
+					.then( () => editor )
+			);
+		} );
+	}
+}
+
+mix( MultirootEditor, DataApiMixin );
+
+/**
+ * The multiroot editor UI class.
+ *
+ * @extends module:core/editor/editorui~EditorUI
+ */
+class MultirootEditorUI extends EditorUI {
+	/**
+	 * Creates an instance of the multiroot editor UI class.
+	 *
+	 * @param {module:core/editor/editor~Editor} editor The editor instance.
+	 * @param {module:ui/editorui/editoruiview~EditorUIView} view The view of the UI.
+	 */
+	constructor( editor, view ) {
+		super( editor );
+
+		/**
+		 * The main (top–most) view of the editor UI.
+		 *
+		 * @readonly
+		 * @member {module:ui/editorui/editoruiview~EditorUIView} #view
+		 */
+		this.view = view;
+
+		/**
+		 * A normalized `config.toolbar` object.
+		 *
+		 * @type {Object}
+		 * @private
+		 */
+		this._toolbarConfig = normalizeToolbarConfig( editor.config.get( 'toolbar' ) );
+	}
+
+	/**
+	 * Initializes the UI.
+	 */
+	init() {
+		const view = this.view;
+		const editor = this.editor;
+		const editingView = editor.editing.view;
+
+		view.render();
+
+		for ( const editable of this.view.editables ) {
+			// The editable UI element in DOM is available for sure only after the editor UI view has been rendered.
+			// But it can be available earlier if a DOM element has been passed to DecoupledEditor.create().
+			const editableElement = editable.element;
+
+			// Register the editable UI view in the editor. A single editor instance can aggregate multiple
+			// editable areas (roots) but the decoupled editor has only one.
+			this._editableElements.set( editable.name, editableElement );
+
+			// Let the global focus tracker know that the editable UI element is focusable and
+			// belongs to the editor. From now on, the focus tracker will sustain the editor focus
+			// as long as the editable is focused (e.g. the user is typing).
+			this.focusTracker.add( editableElement );
+
+			// Let the editable UI element respond to the changes in the global editor focus
+			// tracker. It has been added to the same tracker a few lines above but, in reality, there are
+			// many focusable areas in the editor, like balloons, toolbars or dropdowns and as long
+			// as they have focus, the editable should act like it is focused too (although technically
+			// it isn't), e.g. by setting the proper CSS class, visually announcing focus to the user.
+			// Doing otherwise will result in editable focus styles disappearing, once e.g. the
+			// toolbar gets focused.
+			editable.bind( 'isFocused' ).to( this.focusTracker );
```

Comment:
Why all roots are marked as focused when just one has focus at a time? This is not the right UX. 

![image](https://user-images.githubusercontent.com/1099479/53817621-231fa000-3f66-11e9-8c60-7ef3e05dfccd.png)


The relation with the FT should be different in this multi–root editor:
* if any editable is focused -> `FT#isFocused === true`
* if an editable is focused, only that editable gets focused styles as per [`EditableUIView`](https://github.com/ckeditor/ckeditor5-ui/blob/master/src/editableui/editableuiview.js#L121-L138)

Reply:
We had some discussion with @pjasiun some time ago about this case - https://github.com/ckeditor/ckeditor5-editor-decoupled/issues/21#issuecomment-448943315 (the outcome was that it is acceptable). However, if it can be easily changed with the current API, I'm in favor of making this change to have only one root marked as focused at the same time.

Reply:
If you want to change it, that's fine, but TBH I like it this way. It stresses the fact that this is a single editor.